python/src/main/python/drivers/piglit-worker.py

```diff
@@ -0,0 +1,418 @@
+            res.log += f.read()
+
+    if os.path.isfile(png_file):
+        with gfuzz_common.open_bin_helper(png_file, 'rb') as f:
+            res.PNG = f.read()
+
+    if os.path.isfile(status_file):
+        with gfuzz_common.open_helper(status_file, 'r') as f:
+            status = f.read().rstrip()
+        if status == 'SUCCESS':
+            res.status = tt.JobStatus.SUCCESS
+        elif status == 'CRASH':
+            res.status = tt.JobStatus.CRASH
+        elif status == 'TIMEOUT':
+            res.status = tt.JobStatus.TIMEOUT
+        elif status == 'UNEXPECTED_ERROR':
+            res.status = tt.JobStatus.UNEXPECTED_ERROR
+        else:
+            res.log += '\nUnknown status value: ' + status + '\n'
+            res.status = tt.JobStatus.UNEXPECTED_ERROR
+    else:
+        # Not even a status file?
+        res.log += '\nNo STATUS file\n'
+        res.status = tt.JobStatus.UNEXPECTED_ERROR
+
+    return res
+
+
+def run_image_job(frag_file: str, json_file: str, status_file: str,
+                  png_file: str, output_dir: str, skip_render: bool):
+    """
+    Runs an image job. Converts the shader job to a piglit shader_test file, then delegates to
+    run_shader_test to render with shader_runner. Writes the status of the job to file.
+    :param frag_file: The fragment shader to convert and run.
+    :param json_file: The JSON uniforms to use with the shader.
+    :param status_file: The status file to write to.
+    :param png_file: The PNG file to write to.
+    :param output_dir: The directory to use for the job.
+    :param skip_render: whether to skip rendering or not.
+    """
+
+    assert os.path.isdir(output_dir)
+    assert os.path.isfile(frag_file)
+    assert os.path.isfile(json_file)
+
+    # graphicsfuzz_piglit_converter has to skip the first argument since it's meant to be a
+    # standalone script taking arguments from a command line.
+    arglist = list('')
+    arglist.append(str(json_file))
+    if skip_render:
+        arglist.append(NODRAW_ARG)
+
+    shader_test_file = graphicsfuzz_piglit_converter.get_shader_test_from_job(json_file)
+
+    try:
+        log('Creating shader_test file...')
+        graphicsfuzz_piglit_converter.main_helper(arglist)
+    except Exception as ex:
+        log('Could not create shader_test from the given job.')
+        raise ex
+
+    status = STATUS_SUCCESS
+
+    try:
+        run_shader_test(shader_test_file, skip_render)
+    except subprocess.TimeoutExpired:
+        status = STATUS_TIMEOUT
+    except subprocess.CalledProcessError:
+        status = STATUS_CRASH
+
+    # Piglit throws the output PNG render into whatever the current working directory is
+    # (and there's no way to specify a location to write to) - we need to move it to wherever our
+    # output is.
+
+    time.sleep(2)
+
+    if os.path.isfile(os.getcwd() + '/' + PNG_FILENAME):
+        shutil.move(os.getcwd() + '/' + PNG_FILENAME, os.getcwd() + '/' + png_file)
```

Comment:
Please remove `os.getcwd()` and `/`. You should use `os.path.join('a', 'b', 'c')` to create a path, but you don't need to here anyway; you can just use `PNG_FILENAME` and `png_file`.

Reply:
Used `os.getcwd()` because I thought it would resolve an unrelated error when debugging this - thanks for catching.